ark/phenotyping/cluster_test.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import pandas as pd
+
+import ark.phenotyping.cluster as cluster
+
+
+def test_winner():
+    test_sample = np.array([0.01, 0.02])
+    test_weights = np.array([[[0.001, 0.005],
+                              [0.002, 0.004]],
+                             [[0.003, 0.009],
+                              [0.008, 0.027]]])
+
+    winning_coords = cluster.winner(test_sample, test_weights)
+
+    assert winning_coords == (1, 1)
+
+
+def test_update():
+    test_sample = np.array([0.01, 0.02])
+    test_weights = np.array([[[0.001, 0.005],
+                              [0.002, 0.004]],
+                             [[0.003, 0.009],
+                              [0.008, 0.027]]])
+    test_winning_coords = (1, 1)
+    test_sigma = 1.0
+    test_learning_rate = 0.5
+    test_x_mesh, test_y_mesh = np.meshgrid(np.arange(2), np.arange(2))
+
+    test_t = 1
+    test_num_iters = 1
+
+    # it's impractical to check exact values, so we'll round to make life easier
+    weights = cluster.update(test_sample, test_weights, test_winning_coords,
+                             test_sigma, test_learning_rate, test_x_mesh, test_y_mesh,
+                             test_t, test_num_iters)
+    weights = np.round(weights, decimals=8)
+
+    result = np.array([[[0.00108549, 0.00514249],
```

Comment:
Can we also check that weights of the matching neuron are the ones that are changed the most?